### DISCUSSION TITLE
Prioritize one queue over another?

### DISCUSSION THREAD

**Discussion by stim371** [OptimalBits/bull#2453] [Q&A]
Is it possible to prioritize one queue over another?

I see there is the ability to add a priority to a job, which orders it inside of that specific queue, but nest's ProcessOptions only have name/concurrency as options.

I am implementing several data backfill jobs that will churn away at a large number of jobs, but I would like the async processing for customer requests to have top priority so they have a good experience and don't get bogged down in thousands of jobs that could easily wait their turn.

Coming from ruby-land, I could configure each queue with its own priority to achieve this (https://github.com/mperham/sidekiq/wiki/Advanced-Options#queues). Is there something similar in Bull? (preferably without spinning up more worker processes)

**Comment by manast** [comment]
@stim371 It seems that the requirement you have will be solved by the Pro version of BullMQ (https://docs.bullmq.io/bullmq-pro/groups). I know many people use the NestJS plugin for Bull, but I am puzzled about what it provides that cannot be easily accomplished by using the API provided by BullMQ directly...
